plugin/storage/es/samplingstore/storage.go

```diff
@@ -0,0 +1,199 @@
+	}
+	var retSamples []dbmodel.Throughput
+	hits := searchResult.Hits.Hits
+	for _, hit := range hits {
+		source := hit.Source
+		var tToD dbmodel.TimeThroughput
+		if err := json.Unmarshal(*source, &tToD); err != nil {
+			return nil, fmt.Errorf("unmarshalling documents failed: %w", err)
+		}
+		retSamples = append(retSamples, tToD.Throughput...)
+	}
+	return dbmodel.ToThroughputs(retSamples), nil
+}
+
+func (s *SamplingStore) InsertProbabilitiesAndQPS(hostname string,
+	probabilities model.ServiceOperationProbabilities,
+	qps model.ServiceOperationQPS,
+) error {
+	ts := time.Now()
+	writeIndexName := indexWithDate(s.samplingIndexPrefix, s.indexDateLayout, ts)
+	val := dbmodel.ProbabilitiesAndQPS{
+		Hostname:      hostname,
+		Probabilities: probabilities,
+		QPS:           qps,
+	}
+	s.writeProbabilitiesAndQPS(writeIndexName, ts, val)
+	return nil
+}
+
+func (s *SamplingStore) GetLatestProbabilities() (model.ServiceOperationProbabilities, error) {
+	ctx := context.Background()
+	indices, err := s.getLatestIndices(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get latest indices: %w", err)
+	}
+	searchResult, err := s.client().Search(indices...).
+		Size(s.maxDocCount).
+		IgnoreUnavailable(true).
+		Do(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to search for Latest Probabilities: %w", err)
+	}
+	lengthOfSearchResult := len(searchResult.Hits.Hits)
+	if lengthOfSearchResult == 0 {
+		return nil, nil
+	}
+
+	var latestProbabilities dbmodel.TimeProbabilitiesAndQPS
+	latestTime := time.Time{}
+	for _, hit := range searchResult.Hits.Hits {
+		var data dbmodel.TimeProbabilitiesAndQPS
+		if err = json.Unmarshal(*hit.Source, &data); err != nil {
+			return nil, fmt.Errorf("unmarshalling documents failed: %w", err)
+		}
+		if data.Timestamp.After(latestTime) {
+			latestTime = data.Timestamp
+			latestProbabilities = data
+		}
+	}
+	return latestProbabilities.ProbabilitiesAndQPS.Probabilities, nil
+}
+
+func (s *SamplingStore) writeThroughput(indexName string, ts time.Time, throughputs []*model.Throughput) {
+	s.client().Index().Index(indexName).Type(throughputType).
+		BodyJson(&dbmodel.TimeThroughput{
+			Timestamp:  ts,
+			Throughput: dbmodel.FromThroughputs(throughputs),
+		}).Add()
+}
+
+func (s *SamplingStore) writeProbabilitiesAndQPS(indexName string, ts time.Time, pandqps dbmodel.ProbabilitiesAndQPS) {
+	s.client().Index().Index(indexName).Type(probabilitiesType).
+		BodyJson(&dbmodel.TimeProbabilitiesAndQPS{
+			Timestamp:           ts,
+			ProbabilitiesAndQPS: pandqps,
+		}).Add()
+}
+
+func (s *SamplingStore) getLatestIndices(ctx context.Context) ([]string, error) {
+	now := time.Now().UTC()
+	for {
+		indexName := indexWithDate(s.samplingIndexPrefix, s.indexDateLayout, now)
+		exists, err := s.client().IndexExists(indexName).Do(ctx)
+		if err != nil {
+			return nil, err
+		}
+		if exists {
+			return []string{indexName}, nil
+		}
+		now = now.Add(-time.Hour * 24)
```

Comment:
this is not exactly what I had in mind. First, this loop can be infinite on an empty database. Second, 24hrs is pretty arbitrary; we don't know how often users rotate indices, so T-24h could skip the latest index and go to earlier one.

My suggestion was to have a user-provided configuration parameter how far back the search should go.